rtengine/imageio.cc

```diff
@@ -83,6 +85,43 @@ Glib::ustring to_utf8 (const std::string& str)
     }
 }
 
+unsigned short addSat (unsigned short a, unsigned short b)
```

Comment:
@heckflosse The whole loop could well be manually vectorized with `_mm_adds_pu16`. I didn't have a look at the assemby yet, but my guess is, the auto-vectorizer came to the same conclusion.